packages/rest/test/integration/http-handler.integration.ts

```diff
@@ -393,6 +450,83 @@ describe('HttpHandler', () => {
     }
   });
 
+  context('multipart/form-data', () => {
```

Comment:
As I mentioned before, IMO the test for multipart is rather complex and unnecessarily complex for integration tests. I would like to see an acceptance-level test for this feature.

Also, I don't think we should be telling LB4 users to handle file uploads by calling `multer` directly from the controller. Instead, we should teach them how to build a custom body parser using multer under the hood.

Reply:
This is a just a starting point to provide `multipart/form-data` integration.

Reply:
I think we may not understand each other.

As I understand this pull request, it's adding the following two extension points:
- pluggable body parsers
- consume request body as a stream
while using "file upload" scenario as an acceptance criteria to verify the implementation.

I am asking you to:
1. Implement the simplest possible integration test for consuming request body as a stream. One that does not rely on complex stuff like multipart processing. When this integration test fail, it must be easy to identify the source of the problem. With multipart in play, it's difficult to tell whether the problem is in our framework or in the interaction with multer.

2. Write an acceptance test to show how this new feature can be used to implement file uploads. Essentially, move the test code inside `context('multipart/form-data')` into a new file in the [test/acceptance](`https://github.com/strongloop/loopback-next/tree/b090e0706c451df57deebc2bd92595787e0606d2/packages/rest/test/acceptance) directory and beef little bit up to work at RestApplication level.

Reply:
I am totally fine to keep `multipart/form-data` integration very minimal in this pull request. I actually love it how this pull request is reframing file-upload support from a built-in feature (as proposed in https://github.com/strongloop/loopback-next/pull/1989) into an acceptance test serving as a possible reference implementation allowing anybody to implement any custom file-upload handing.

Reply:
As I am thinking about this more, it would be even more cool if we could have two acceptance tests for file upload:
- one showing how to handle file uploads by registering a custom request body parser
- the other showing how to handle file uploads in individual controller methods by receiving the request body as a stream.